lib/soci/src/backends/sqlite3/session.cpp

```diff
@@ -59,7 +59,8 @@ sqlite3_session_backend::sqlite3_session_backend(
     int connection_flags = SQLITE_OPEN_READWRITE | SQLITE_OPEN_CREATE;
     std::string synchronous;
     std::string const & connectString = parameters.get_connect_string();
-    std::string dbname(connectString);
+    std::string& dbname = db_name_;
```

Comment:
I don't think the changes in this commit are necessary, you can use https://www.sqlite.org/pragma.html#pragma_database_list instead. I've tested it locally from the `sqlite3` command line interface, and it does return the path to the connected database.

I'm not _strongly_ opposed to making this change, but I do think it's better to avoid changing dependencies whenever possible to make it easier to upgrade in the future.

Reply:
ah yes you are right - I missed it when I looked at the list of pragmas. Updated to use it instead